### PR TITLE
[Theme Adapter] Migrate all theme adapter libraries to Accompanist

### PR DESCRIPTION
**Pull request by ricknout**
This migrates and merges all theme adapter libraries into Accompanist (source code, tests, docs, etc). There has also been a refactor to centralize all theme adapter logic into a core module, which is used by the AppCompat, Material, and Material 3 modules.

These libraries include:
* **AppCompat Theme Adapter**: Migrated [`com.google.accompanist:accompanist-appcompat-theme`](https://github.com/google/accompanist/tree/main/appcompat-theme) to `com.google.accompanist:accompanist-themeadapter-appcompat`
* **Material Theme Adapter**: Migrated [`com.google.android.material:compose-theme-adapter`](https://github.com/material-components/material-components-android-compose-theme-adapter#compose-material) to `com.google.accompanist:accompanist-themeadapter-material`
* **Material 3 Theme Adapter**: Migrated [`com.google.android.material:compose-theme-adapter-3`](https://github.com/material-components/material-components-android-compose-theme-adapter#compose-material-3) to `com.google.accompanist:accompanist-themeadapter-material3`
* **Core Theme Adapter**: Migrated [`com.google.android.material:compose-theme-adapter-core`](https://maven.google.com/web/index.html#com.google.android.material:compose-theme-adapter-core) to `com.google.accompanist:accompanist-themeadapter-core`

Once merged, a follow-up PR will be made against the [MDC-Android Compose Theme Adapter repository](https://github.com/material-components/material-components-android-compose-theme-adapter) to deprecate those libraries in favor of these.

### CODE REVIEW

File: sample/src/main/java/com/google/accompanist/sample/themeadapter/Material3Sample.kt [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun Material3Sample() {
+    val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior(rememberTopAppBarState())
+    Scaffold(
+        modifier = Modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
+        /*topBar = {
```
Comment (florina-muntenescu) [inline sample/src/main/java/com/google/accompanist/sample/themeadapter/Material3Sample.kt:105]: why is this commented out? can it be added or removed?

File: sample/src/main/java/com/google/accompanist/sample/themeadapter/MaterialSample.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+fun MaterialSamplePreview() {
+    MdcTheme {
+        MaterialSample()
+    }
+}
+
+@Composable
+fun MaterialSample() {
+    Scaffold(
+        /*topBar = {
```
Comment (florina-muntenescu) [inline sample/src/main/java/com/google/accompanist/sample/themeadapter/MaterialSample.kt:71]: why is this commented out?

if we want to show that this is optional, maybe we can add a comment that says optional?

File: .github/ISSUE_TEMPLATE/theme-adapter-material-bug-report copy.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (bentrengrove) [inline .github/ISSUE_TEMPLATE/theme-adapter-material-bug-report copy.md:1]: File name needs changing

File: themeadapter-appcompat/gradle.properties [hunk 4]
```diff
@@ -14,6 +14,6 @@
 # limitations under the License.
 #
 
-POM_ARTIFACT_ID=accompanist-appcompat-theme
+POM_ARTIFACT_ID=accompanist-themeadapter-appcompat
```
Comment (bentrengrove) [inline themeadapter-appcompat/gradle.properties:17]: Won't this mess people up upgrading? Do we have to have a duplicate deprecated version?

Comment (ricknout) [reply themeadapter-appcompat/gradle.properties:17]: TBH I'm not entirely sure, I'll set up some time for us to chat through the migration path

Comment (ricknout) [reply themeadapter-appcompat/gradle.properties:17]: Added back the original artifact (with deprecation/migration) as discussed

**Review by bentrengrove — APPROVED** [review]
LGTM!

**Pull request merged**